src/glossary.adoc

```diff
@@ -0,0 +1,116 @@
+[[sec:glossary]]
+[glossary]
+= Glossary
+
+[glossary]
+
+arity::
+The number of arguments declared by or passed to a macro.  A macro can be _fixed_ or _variable_
+arity; the number of subforms in an invocation is its _actual arity_.
+
+argument::
+A single subform within an encoding expression, corresponding to one of the macro’s parameters.
+This is a one-to-one relation except in the case of rest parameters, where any number of trailing
+arguments are passed to the same parameter.
```

Comment:
I don't think this is quite right givven our current vernacular.  Isn't an argument a sub-expression bound to a parameter, and each parameter may have zero or more arguments depending on its _argument cardinality_?

Reply:
There's an unanswered question lurking in this terminology.
```
(public mac [(any* p), (any... q)] /*template*/)
(:mac [elt1, elt2] elt3 elt4)
```
Are `elt1` and `elt2` each arguments, or is `[elt1, elt2]` the argument?  If the former is correct, what do we call the latter?  And what about the rest-arguments?

I've failed to coalesce on terminology for all the parts here.

Reply:
Here is my current thinking (which may not be right):

* `elt1`, `elt2`, `elt3` are all _arguments_.  `elt1`, for example is a single _argument_.
* `[elt1, elt2]` (the grouping) are the _arguments_ for a _parameter_.  I think this could be shorthanded as _parameter arguments_ or maybe _parameter argument group_.

Reply:
Cool, that's aligned with my thinking.  I like "argument group" a lot, and need to see how it flows with the related narratives.